### DISCUSSION TITLE
date picker component

### DISCUSSION THREAD

**Discussion by IamGauravSaini** [cloudscape-design/components#2431] [Q&A] [answered]
i want to use date format dd/mm/yyyy but the component is not allowing can you help me with it
how can i modify it to dd/mm/yyyy format

**Comment by johannes-weber** [comment]
Hi @IamGauravSaini, Thanks for reaching out!

This seems like a reasonable request. Could you share you use-case for having more context about the "why" behind?

Best, Johannes

**Comment by gethinwebster** [comment]
We would not recommend using the format `dd/mm/yyyy` as it is commonly confused (e.g. does `01/02/2024` mean 1st February or 2nd January?). That's why the current format of yyyy/mm/dd for DatePicker was chosen, in order to be a single internationally understood format.

**Comment by pmosconi** [comment]
Hi, many applications need not be internationalized but live in a specific locale: users keep asking to "fix" the date picker format.
Please reconsider your position and leave the choice to us.
Thanks

**Comment by louhibi** [comment]
Yes this is very annoying for users in europe and more than half the globe actually, this is standard only in North America. and very confusing for users from other parts of the world.

**Reply by SpyZzey** [reply]
Hey,
The DatePicker accepts a `format`-property. This can be set to `long-localized` to show a localized date string (e.g., "January 1, 2026" in en-US locale or "1 January 2026" in en-GB locale). Does this work for your use-case or do you need a more numeric format (e.g., 01/01/2026)?

**Reply by louhibi** [reply]
Hey @SpyZzey,

Thanks for looking into this.

The issue I’m seeing with European users mainly happens with input, since the display side is already fixed. When users don’t use the DatePicker and instead type directly into the input field, they encounter the non-localized format.

Because everything else in the UI is localized, this feels inconsistent and ends up being surprising for them.

**Comment by SpyZzey** [comment, marked as answer]
Hey,
Good news! 
We have discussed this internally and we want to move forward with allowing custom date formats for more flexibility. 

**Note**: This will be tracked internally and we don't have an estimate on when this will be released yet.

**Reply by louhibi** [reply]
Than you 🙏

**Reply by louhibi** [reply]
My suggestion is to just format it according to the locale prop. But custom format is an ok solution Also.

**Reply by SpyZzey** [reply]
The final API and implementation details are still open for discussion. 
Feel free to share insights on your use cases so we can keep that in mind.
